### PR TITLE
Fix / rewrite xfail tests in docs

### PR DESCRIPTION
**Pull request by adriangb**
Selected Reviewer: @dmontagu

### CODE REVIEW

File: docs/usage/models.md [hunk 1]
```diff
@@ -713,29 +713,28 @@ class DataModel(BaseModel):
...
     data: Optional[DataT] = None
-    error: Optional[Error] = Field(validate_default=True)
-
-    @field_validator('error')
-    def check_consistency(cls, v, info):
-        if v is not None and info.data['data'] is not None:
-            raise ValueError('must not provide both data and error')
-        if v is None and info.data.get('data') is None:
-            raise ValueError('must provide data or error')
-        return v
```
Comment (adriangb) [inline docs/usage/models.md:724]: This validate default stuff doesn't have anything to do with generic models (the docs section). I'm simplifying the example by removing it.

File: docs/usage/schema.md [hunk 2]
```diff
@@ -695,52 +695,52 @@ class Model(BaseModel):
...
-
-print(Person.schema_json(indent=2))
-```
-
-
-For more fine-grained control, you can alternatively set `schema_extra` to a callable and post-process the generated schema.
-The callable can have one or two positional arguments.
-The first will be the schema dictionary.
-The second, if accepted, will be the model class.
-The callable is expected to mutate the schema dictionary *in-place*; the return value is not used.
```
Comment (adriangb) [inline docs/usage/schema.md:738]: Since it now always has to be a callable having two sections doesn't make sense.

**Comment by adriangb** [comment]
please review

File: docs/usage/models.md (hunk 1 above)
Comment (dmontagu) [reply docs/usage/models.md:724]: I liked this example for the sake of showing how to build a type with a "common"/"universal" error field, and generic "data" field, as one approach to getting consistent error handling in a web API. (Essentially functioning as a field-name-tagged union.) (I vaguely recall having written this example myself, maybe I'm misremembering..)

I liked the example as I felt like it showed a bit more complex/"real world" usage. But perhaps the complexity distracts from the main point.

Either way, I think we should eventually make a point of having more "intermediate" level examples somewhere in the docs. But I guess that's something @tpdorsey will be tracking.

**Review by dmontagu — APPROVED** [review]

**Pull request merged**